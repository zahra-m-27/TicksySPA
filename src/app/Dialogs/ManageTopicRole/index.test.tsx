import ReactDOM from 'react-dom';
import React, {useRef} from 'react';
import {render} from '@testing-library/react';
import renderer from 'react-test-renderer';
import ManageTopicRoleDialog from './index';
import user from '@testing-library/user-event';

const Sample: React.FC = ({}) => {
  const ref = useRef(() => {});
  return <ManageTopicRoleDialog topicId={0} onDismissRef={ref} />;
};

const sample = <Sample />;

test('render in ReactDOM', () => {
  const div = document.createElement('div');
  ReactDOM.render(sample, div);
});

it('render header', () => {
  const {getByText} = render(sample);
  getByText(/افزودن نقش/i);
});

it('render role label', () => {
  const {getByText} = render(sample);
  getByText(/عنوان نقش/i);
});

it('render buttons', () => {
  const {getByText} = render(sample);
  getByText(/نقش جدید/i);
  getByText(/ثبت/i);
});

it('open new dialog', async () => {
  const {getByText, findByTestId} = render(sample);
  const newRoleButton = getByText(/نقش جدید/i);
  user.click(newRoleButton);
  const dialogContainer = await findByTestId('dialog-container');
  user.click(dialogContainer);
});

it('snapshot', () => {
  const snapshot = renderer.create(sample).toJSON();
  expect(snapshot).toMatchSnapshot();
});
