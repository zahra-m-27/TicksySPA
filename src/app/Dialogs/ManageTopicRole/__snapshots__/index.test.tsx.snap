// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="container"
>
  <div
    className="header"
  >
    <span>
      افزودن نقش
    </span>
    <img
      alt="add"
      src="add_user_menu.svg"
    />
  </div>
  <div
    className="content"
    style={
      Object {
        "visibility": undefined,
      }
    }
  >
    <img
      alt="add"
      className="avatar"
      src="add_user.svg"
    />
    <div
      className="container"
    >
      <label
        className="label"
      >
        عنوان نقش
      </label>
      <div
        className="input_container"
        data-testid="input-container"
      >
        <input
          className="input_class"
          dir="auto"
          onChange={[Function]}
          onKeyDown={[Function]}
          value=""
        />
      </div>
    </div>
    <div
      className="buttons"
    >
      <div
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": undefined,
          }
        }
      >
        <span>
          نقش جدید
        </span>
      </div>
      <div
        className="button"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#1354ac",
          }
        }
      >
        <span>
          ثبت نقش
        </span>
      </div>
    </div>
  </div>
  <div
    className="ant-spin ant-spin-lg ant-spin-spinning loading"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <span
      className="ant-spin-dot ant-spin-dot-spin"
    >
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
      <i
        className="ant-spin-dot-item"
      />
    </span>
  </div>
</div>
`;
